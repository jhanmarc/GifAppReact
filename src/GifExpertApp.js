import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = [ 'One Punch', 'Samurai X', 'Dragon Ball' ]
    const [categories, setCategories] = useState( defaultCategories );
    
    // const handleAdd = () => {
    //     // setCategories( [...categories, 'Bleach'] );
    //     setCategories( categ => [ ... categ, 'Bleach' ] )
    // }
    
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories } />
            <hr />

            <ol>
                { 
                    categories.map( (category) => 
                       <GifGrid 
                           category= { category }
                           key = { category }
                       />
                    ) 
                }
                <br />
            </ol>
        </div>
    )
}
