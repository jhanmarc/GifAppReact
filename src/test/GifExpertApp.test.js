import React from 'react';

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en <GifExpertApp>', ()=> {

    const categorias = [ 'One Punch', 'Dragon Ball' ];

    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow( <GifExpertApp  /> );
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        
        const wrapper = shallow( <GifExpertApp defaultCategories={ categorias } /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categorias.length );


    })
    
    
})