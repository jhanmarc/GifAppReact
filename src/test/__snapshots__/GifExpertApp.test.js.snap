// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp> debe de mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <br />
  </ol>
</div>
`;

exports[`Pruebas en <GifExpertApp> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <br />
  </ol>
</div>
`;
